crates/rolldown_binding/src/types/binding_rendered_chunk.rs

```diff
@@ -34,13 +39,48 @@ impl From<rolldown_common::RollupRenderedChunk> for RenderedChunk {
       module_ids: value.module_ids.into_iter().map(|x| x.to_string()).collect(),
       exports: value.exports,
       file_name: value.filename.to_string(),
-      modules: value
-        .modules
-        .into_iter()
-        .map(|(key, value)| (key.to_string(), value.into()))
-        .collect(),
+      modules: BindingChunkModules::new(value.modules),
       imports: value.imports.iter().map(|x| x.to_string()).collect(),
       dynamic_imports: value.dynamic_imports.iter().map(|x| x.to_string()).collect(),
     }
   }
 }
+
+// use own map wrapper to workaround "\0" issue
+// https://github.com/napi-rs/napi-rs/blob/f116eaf5e54090db4dca8a00ccdb684543a39e86/crates/napi/src/bindgen_runtime/js_values/map.rs#L26
```

Comment:
Wonder if we should enable napi's `experimental` feature

https://github.com/napi-rs/napi-rs/blob/f116eaf5e54090db4dca8a00ccdb684543a39e86/crates/napi/src/bindgen_runtime/js_values/map.rs#L31-L38C11

So it solves the problem under the hood.

@Brooooooklyn cc